src/target/riscv/riscv.c

```diff
@@ -1880,11 +1874,21 @@ static int riscv_resume(
 		targets = &single_target_list;
 	}
 
+	LOG_TARGET_DEBUG(target, "current=%s, address=0x%"
+				TARGET_PRIxADDR ", handle_breakpoints=%s, debug_exec=%s",
+				current ? "true" : "false",
+				address,
+				handle_breakpoints ? "true" : "false",
+				debug_execution ? "true" : "false");
+
 	struct target_list *tlist;
 	foreach_smp_target_direction(resume_order == RO_NORMAL, tlist, targets) {
 		struct target *t = tlist->target;
+		LOG_TARGET_DEBUG(t, "target->state=%s", target_state_name(t));
 		if (t->state != TARGET_UNAVAILABLE) {
-			if (resume_prep(t, current, address, handle_breakpoints,
+			if (t->state != TARGET_HALTED)
+				LOG_TARGET_DEBUG(t, "skipping this target: target not halted");
+			else if (resume_prep(t, current, address, handle_breakpoints,
```

Comment:
Since this is just a debug message, you can also print it out when the hart is unavailable, simplifying the code.

Reply:
Addressed.